float64 vx
float64 vy
